src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,412 @@
+        eResponse.set_creationdate(std::to_string(session->getCreationDate()));
+        eResponse.set_id(session->getSessionID());
+        eResponse.set_lifespan(session->getLifespan());
+        eResponse.set_policyname(session->getPolicyName());
+        eResponse.set_routename(session->getRouteName());
+
+        eResponse.set_status(eEngine::ReturnStatus::OK);
+
+        return toWazuhResponse(eResponse);
+    };
+}
+
+api::Handler sessionsGet(void)
+{
+    return [](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::SessionsGet_Request;
+        using ResponseType = eTest::SessionsGet_Response;
+        auto res = fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        const auto& eRequest = std::get<RequestType>(res);
+
+        auto& sessionManager = SessionManager::getInstance();
+        auto list = sessionManager.getSessionsList();
+
+        ResponseType eResponse;
+        for (const auto& sessionName : list)
+        {
+            eResponse.add_list(sessionName);
+        }
+        eResponse.set_status(eEngine::ReturnStatus::OK);
+
+        return toWazuhResponse(eResponse);
+    };
+}
+
+api::Handler sessionsDelete(shared_ptr<::router::Router> router, shared_ptr<catalog::Catalog> catalog)
+{
+    return [router, catalog](api::wpRequest wRequest) -> api::wpResponse
+    {
+        using RequestType = eTest::SessionsDelete_Request;
+        using ResponseType = eTest::SessionsDelete_Response;
+        auto res = fromWazuhRequest<RequestType, ResponseType>(wRequest);
+
+        // If the request is not valid, return the error
+        if (std::holds_alternative<api::wpResponse>(res))
+        {
+            return std::move(std::get<api::wpResponse>(res));
+        }
+
+        const auto& eRequest = std::get<RequestType>(res);
+
+        const auto errorMsg =
+            !eRequest.has_name() && !eRequest.has_removeall()
+                ? std::make_optional("Missing both /name and /removeall fields, at least one field must be set")
+                : std::nullopt;
+        if (errorMsg.has_value())
+        {
+            return genericError<ResponseType>(errorMsg.value());
+        }
+
+        auto& sessionManager = SessionManager::getInstance();
+
+        if (eRequest.has_removeall() && eRequest.removeall())
+        {
+            for (auto& sessionName : sessionManager.getSessionsList())
+            {
+                const auto deleteSessionError = deleteSession(sessionName, router, catalog);
+                if (deleteSessionError.has_value())
+                {
+                    return genericError<ResponseType>(deleteSessionError.value().message);
+                }
+            }
+        }
+        else if (eRequest.has_name())
```

Comment:
**thought:** are we here sure that at this point that both conditions `eRequest.has_removeall() && eRequest.removeall()` are false? What if just `eRequest.has_removeall()` is false, does the else clause make sense? Please disregard if both conditions are semantically linked.

Reply:
If `eRequest.has_removeall()` is false, then `eRequest.removeall()` does not exist